src/compiler/clang.rs

```diff
@@ -73,6 +77,23 @@ impl CCompilerImpl for Clang {
     where
         T: CommandCreatorSync,
     {
+        let mut ignorable_whitespace_flags = vec!["-P".to_string()];
+
+        // Clang 14 and later support -fminimize-whitespace, which normalizes away whitespace which in turn increases cache hit rate.
+        if !self.is_appleclang {
+            if let Some(version_val) = self.version.clone() {
+                if let Some(parsed_version) = version_val.split(' ').find(|x| x.contains('.')) {
+                    if let Some(major_version) = parsed_version.split('.').next() {
+                        if let Ok(parsed_major_version) = major_version.parse::<usize>() {
+                            if parsed_major_version >= 14 {
+                                ignorable_whitespace_flags.push("-fminimize-whitespace".to_string())
+                            }
+                        }
+                    }
+                }
```

Comment:
It might be worth to use a specialized crate such as `semver` rather than to parse it manually.

Reply:
Using semver in https://github.com/mozilla/sccache/pull/1162/commits/ba0a922c0cb10a955c2335e78ae7516836c88a66. Also refactored-out the check function.